include/phoenix/vm.hh

```diff
@@ -334,6 +336,20 @@ namespace phoenix {
 		[[nodiscard]] PHOENIX_API const std::string& pop_string();
 		[[nodiscard]] PHOENIX_API std::tuple<symbol*, std::uint8_t, std::shared_ptr<instance>> pop_reference();
 
+		[[nodiscard]] PHOENIX_API std::int32_t
+		get_int(std::shared_ptr<instance>& context,
+		        std::variant<int32_t, float, symbol*, std::shared_ptr<instance>>& value,
+		        uint16_t index);
+		[[nodiscard]] PHOENIX_API float
+		get_float(std::shared_ptr<instance>& context,
+		          std::variant<int32_t, float, symbol*, std::shared_ptr<instance>>& value,
+		          uint16_t index);
+
+		PHOENIX_API void set_int(std::shared_ptr<instance>& context, symbol* ref, uint16_t index, std::int32_t value);
+		PHOENIX_API void set_float(std::shared_ptr<instance>& context, symbol* ref, uint16_t index, float value);
+		PHOENIX_API void
+		set_string(std::shared_ptr<instance>& context, symbol* ref, uint16_t index, std::string_view value);
+
```

Comment:
I'm probably missing something here: Why are we moving the symbol accesses into a separate function? To me it looks like nothing has changed about the actual logic.

Reply:
This was introduces, when solution was callback based, yet decide to keep it - this helps avoids code duplication.

Reply:
Okay, but they should probably be private or protected then, I think, since they're basically an implementation detail.